control-plane/addMissingIds.cpp

```diff
@@ -0,0 +1,214 @@
+#include "addMissingIds.h"
+
+#include "frontends/p4/evaluator/evaluator.h"
+#include "p4RuntimeAnnotations.h"
+#include "p4RuntimeArchStandard.h"
+
+namespace P4 {
+
+const IR::P4Program* MissingIdAssigner::preorder(IR::P4Program* program) {
+    auto evaluator =  P4::EvaluatorPass(refMap, typeMap);
```

Comment:
why hide the evaluator in this pass instead of running it from the PassManager?

Reply:
If I do not compute this in this specific order, the evaluator will return a nullptr for `getTopLevelBlock`. When I move the evaluator in the pass manager the reference to the top level block is not set yet. 

It is a little awkward, the other way to solve this is to make the Symboltable mutable and move a reference to it around.

Reply:
I would like to not use the evaluator at all, the only reason I currently use it is to preserve the exact functionality as in `p4RuntimeSerializer.cpp`.